src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import core.basesyntax.impl.BalanceHandlerImpl;
+import core.basesyntax.impl.PurchaseHandlerImpl;
+import core.basesyntax.impl.ReturnHandlerImpl;
+import core.basesyntax.impl.SupplyHandlerImpl;
+import impl.DataConverterImpl;
+import impl.DataProcessorImpl;
+import impl.ReaderFromFileImpl;
+import impl.ReportCreatorImpl;
+import impl.WriterToFileImpl;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import model.Operation;
+import service.DataConverter;
+import service.DataProcessor;
+import service.ReaderFromFile;
+import service.ReportCreator;
+import service.WriterToFile;
+import strategy.OperationStrategy;
+
+public class Main {
+    private static final String FILE_NAME = "src/main/resources/During the day.csv";
+    private static final String REPORT_FILE_NAME = "src/main/resources/Report file.csv";
+
+    public static void main(String[] args) {
+        ReaderFromFile readerFromFile = new ReaderFromFileImpl();
+        List<String> dataFromFile = readerFromFile.read(FILE_NAME);
+
+        DataConverter dataConverter = new DataConverterImpl();
+        List<FruitTransaction> fruitTransactions = dataConverter.parseData(dataFromFile);
+
+        OperationStrategy operationStrategy = new OperationStrategy(Map.of(
```

Comment:
Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead. In the `main()` method you can pass the strategy map into the service that chooses the correct strategy for each operation. While you are using the Strategy pattern here, ensure that the instantiation of concrete handlers is not done directly in the `main()` method. Consider using a factory or injector to provide these dependencies.